Client/Frontend/Browser/Playlist/PlaylistMediaInfo.swift

```diff
@@ -0,0 +1,504 @@
+                let offset = dataRequest.currentOffset
+                let length = dataRequest.requestedLength
+                
+                var request = URLRequest(url: originalURL)
+                request.cachePolicy = .reloadIgnoringLocalCacheData
+                request.addValue("bytes=\(offset)-\(max(1, length - 1))", forHTTPHeaderField: "Range")
+                return request
+            }()
+            
+            // Should use URLSessionDataDelegate to stream instead of downloading all chunks at once..
+            self.session.dataTask(with: request) { [weak self] data, response, error in
+                guard let self = self else { return }
+                
+                if let data = data {
+                    self.data.append(data)
+                }
+                
+                self.processPendingRequests()
+                
+            }.resume()
+            
+            dataRequests.append(loadingRequest)
+            return true
+        }
+        
+        return false
+    }
+    
+    func resourceLoader(_ resourceLoader: AVAssetResourceLoader, shouldWaitForRenewalOfRequestedResource renewalRequest: AVAssetResourceRenewalRequest) -> Bool {
+        false
+    }
+    
+    func resourceLoader(_ resourceLoader: AVAssetResourceLoader, didCancel loadingRequest: AVAssetResourceLoadingRequest) {
+        if contentInfoRequest == loadingRequest {
+            contentInfoRequest = nil
+            return
+        }
+        
+        dataRequests.removeAll(where: { loadingRequest == $0 })
+    }
+    
+    func processPendingRequests() {
+        let requestsFulfilled = Set<AVAssetResourceLoadingRequest>(self.dataRequests.compactMap {
+            $0.contentInformationRequest?.contentLength = Int64(self.data.count)
+            $0.contentInformationRequest?.isByteRangeAccessSupported = true
+            
+            if self.haveEnoughDataToFulfillRequest($0.dataRequest!) {
+                $0.finishLoading()
+                return $0
+            }
+            return nil
+        })
+        
+        self.dataRequests.removeAll(where: { requestsFulfilled.contains($0) })
+    }
+    
+    func haveEnoughDataToFulfillRequest(_ dataRequest: AVAssetResourceLoadingDataRequest) -> Bool {
+        let requestedOffset = Int(dataRequest.requestedOffset)
+        let currentOffset = Int(dataRequest.currentOffset)
+        let requestedLength = dataRequest.requestedLength
+        
+        if currentOffset <= self.data.count {
+            let bytesToRespond = min(self.data.count - currentOffset, requestedLength)
+            let data = self.data.subdata(in: Range(uncheckedBounds: (currentOffset, currentOffset + bytesToRespond)))
+            dataRequest.respond(with: data)
+            return self.data.count >= requestedLength + requestedOffset
+        }
+        
+        return false
+    }
+}
+
+extension MediaResourceManager {
+    static func shimURL(_ url: URL) -> URL {
```

Comment:
Some of these methods could be unit tested too

Reply:
None of this is used atm. It is for CarPlay and since that is removed "for now", I decided to keep these around for v2. So I'll unit test them for v2 when I am coding it in the car to make sure it works properly.